test/mocha/portable.js

```diff
@@ -0,0 +1,103 @@
+var assert = require("assert");
+var exec = require("child_process").exec;
+
+describe("portable", function() {
+    var readFileBackup;
+    var writeFileBackup;
+    var simpleGlobBackup;
+
+    before(function(done) {
+        this.timeout(5000);
+        var uglifyjscmd = '"' + process.argv[0] + '" bin/uglifyjs';
+        var command = uglifyjscmd + ' --self -cm --wrap minifier';
+
+        assert.strictEqual(global.minifier, undefined);
+
+        exec(command, function (err, stdout) {
+            if (err) {
+                this.skip();
+                throw err;
+            }
+
+            eval(stdout);
+
+            assert.strictEqual(typeof minifier, 'object');
+            assert.strictEqual(minifier, global.minifier);
+            assert.strictEqual(true, minifier.parse('foo;') instanceof minifier.AST_Node);
+
+            readFileBackup = minifier.readFile;
+            writeFileBackup = minifier.writeFile;
+            simpleGlobBackup = minifier.simple_glob;
+
+            done();
+        });
+    });
+
+    beforeEach(function() {
+        assert(typeof minifier, "object");
+
+        minifier.readFile = readFileBackup;
+        minifier.writeFile = writeFileBackup;
+        minifier.simple_glob = simpleGlobBackup;
+    });
+
+    after(function() {
+        global.minifier = undefined;
+        assert.strictEqual(global.minifier, undefined);
+    });
+
+    it("Should minify from a string successfully", function() {
+        assert.strictEqual(minifier.minify('console.log("Hello world!");', {fromString: true}).code,
```

Comment:
Suggest minifying `'console["log"]("Hello " + "world!");'` in all your examples in this file to show that minify() actually did something.